libs/thread_pools/include/hpx/thread_pools/scheduled_thread_pool_impl.hpp

```diff
@@ -1776,6 +1776,21 @@ namespace hpx { namespace threads { namespace detail {
             thread_count_.load();
     }
 
+    template <typename Scheduler>
+    std::int64_t scheduled_thread_pool<Scheduler>::get_idle_core_count() const
+    {
+        std::int64_t count = 0;
+        std::size_t i = 0;
+        for (auto const& data : counter_data_)
+        {
+            if (!data.tasks_active_ && sched_->Scheduler::is_core_idle(i++))
```

Comment:
Uhh, I see now that this is not correct. This code will increment `i` only if the current core is not active.

Reply:
See #4643 for a fix.